### PR TITLE
Add support for Hub XMLRPC API container

### PR DESCRIPTION
**Pull request by nadvornik**
## What does this PR change?

This PR add support for Hub XMLRPC API container.
It creates the systemd service on install and handles it on start, stop, restart,status and uninstall commands.

It includes also a bugfix - restart of postgres after installing new certificate - this is needed for reportdb access from hub.

The service depends on https://github.com/uyuni-project/uyuni/pull/8710

## Test coverage
- No tests: **add explanation**

- [ ] **DONE**

## Links

https://github.com/uyuni-project/uyuni-tools/issues/292
https://github.com/uyuni-project/uyuni/pull/8710

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by cbosdo — COMMENTED** [review]
We would also need to handle the Hub XML-RPC API container for the kubernetes case. This would surely require changes in the server helm chart.

File: shared/utils/volumes.go [hunk 1]
```diff
@@ -97,6 +97,11 @@ var ServerVolumes = append([]types.Volume{
 	{Name: "ca-cert", PersistentVolumeClaim: &types.PersistentVolumeClaim{ClaimName: "ca-cert"}},
 }, etcAndPgsqlVolumes[:]...)
 
+// HubXmlrpcVolumeMounts represents volumes used by Hub Xmlrpc container.
+var HubXmlrpcVolumeMounts = []types.VolumeMount{
+	{MountPath: "/etc/pki/trust/anchors", Name: "ca-cert"},
```
[the remark this answers is not in the document]
Comment (nadvornik) [reply shared/utils/volumes.go:102]: This together with https://github.com/uyuni-project/uyuni/pull/8710 is for connection  from the Hub XML-RPC API container to Peripheral servers.
I am not sure about  the SSL key of the Hub XML-RPC API container.

[the remark this answers is not in the document]
Comment (nadvornik) [reply shared/utils/volumes.go:102]: Apparently we never had SSL support for Hub API. Even 4.3 documentation mentions only http to port 2830.

**Comment by nadvornik** [comment]
I have rebased it and added a fix for utils.Errorf. Now the PR should be complete, except for kubernetes support.

File: mgradm/cmd/install/podman/utils.go [hunk 2]
```diff
@@ -47,6 +47,29 @@ func setupCocoContainer(flags *podmanInstallFlags) error {
 	return nil
 }
 
+func setupHubXmlrpcContainer(flags *podmanInstallFlags) error {
+	if flags.HubXmlrpc.Enable {
+		log.Info().Msg(L("Enabling Hub XMLRPC API container."))
```
Comment (cbosdo) [inline mgradm/cmd/install/podman/utils.go:52]:
```suggestion
		log.Info().Msg(L("Enabling Hub XML-RPC API container."))
```

File: mgradm/shared/podman/podman.go [hunk 3]
```diff
@@ -65,6 +65,28 @@ Environment=database_password=%s
...
+func GenerateHubXmlrpcSystemdService(image string) error {
+	hubXmlrpcData := templates.HubXmlrpcServiceTemplateData{
+		Volumes:    utils.HubXmlrpcVolumeMounts,
+		Ports:      utils.HUB_XMLRPC_PORTS,
+		NamePrefix: "uyuni",
+		Network:    podman.UyuniNetwork,
+		Image:      image,
+	}
+	if err := utils.WriteTemplateToFile(hubXmlrpcData, podman.GetServicePath(podman.HubXmlrpcService), 0555, false); err != nil {
+		return fmt.Errorf(L("failed to generate systemd service unit file: %s"), err)
```
Comment (cbosdo) [inline mgradm/shared/podman/podman.go:78]:
```suggestion
		return utils.Errorf(err, L("failed to generate systemd service unit file"))
```

File: mgradm/shared/podman/podman.go [hunk 4]
```diff
@@ -65,6 +65,28 @@ Environment=database_password=%s
...
+	}
+
+	environment := fmt.Sprintf(`Environment=UYUNI_IMAGE=%s
+	`, image)
+	if err := podman.GenerateSystemdConfFile(podman.HubXmlrpcService, "Service", environment); err != nil {
+		return fmt.Errorf(L("cannot generate systemd conf file: %s"), err)
```
Comment (cbosdo) [inline mgradm/shared/podman/podman.go:84]:
```suggestion
		return utils.Errorf(err, L("cannot generate systemd configuration file"))
```

**Review by cbosdo — APPROVED** [review]

**Pull request merged**